velox/connectors/hive/storage_adapters/hdfs/HdfsFileSink.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <hdfs/hdfs.h>
+#include "velox/common/file/FileSystems.h"
+#include "velox/connectors/hive/storage_adapters/hdfs/HdfsWriteFile.h"
+#include "velox/core/Context.h"
+#include "velox/dwio/common/DataSink.h"
+
+namespace facebook::velox {
+
+class HdfsFileSink : public facebook::velox::dwio::common::DataSink {
+ public:
+  explicit HdfsFileSink(
+      const std::string& fullDestinationPath,
+      const facebook::velox::dwio::common::MetricsLogPtr& metricLogger =
+          facebook::velox::dwio::common::MetricsLog::voidLog(),
+      facebook::velox::dwio::common::IoStatistics* stats = nullptr)
+      : facebook::velox::dwio::common::DataSink{
+            "HdfsFileSink",
+            metricLogger,
+            stats} {
+    // Extract the host and port.
+    auto destinationPathStartPos = fullDestinationPath.substr(7).find("/", 0);
+    std::string destinationPath =
+        fullDestinationPath.substr(destinationPathStartPos + 7);
+    std::string hostAndPort =
+        fullDestinationPath.substr(7).substr(0, destinationPathStartPos);
+    auto portStartPos = hostAndPort.find(":", 0);
+    std::string host = hostAndPort.substr(0, portStartPos);
+    std::string port = hostAndPort.substr(portStartPos + 1);
+    std::unordered_map<std::string, std::string> configurationValues(
+        {{"hive.hdfs.host", host}, {"hive.hdfs.port", port}});
+    auto memConfig =
+        std::make_shared<const core::MemConfig>(configurationValues);
+    std::string hdfsFilePath = "hdfs://" + host + ":" + port + destinationPath;
+    auto hdfsFileSystem = filesystems::getFileSystem(hdfsFilePath, memConfig);
```

Comment:
no need to pass memConfig here as getFileSystem can parse the host/port from Path
```suggestion
    auto hdfsFileSystem = filesystems::getFileSystem(hdfsFilePath, nullptr);
```